cumulusci/tasks/github/merge.py

```diff
@@ -15,57 +15,187 @@ class MergeBranch(BaseGithubTask):
         'branch_prefix': {
             'description': "The prefix of branches that should receive the merge.  Defaults to project__git__prefix_feature",
         },
+        'children_only': {
+            'description': "If True, merge will only be done to child branches.  This assumes source branch is a parent feature branch.  Defaults to False",
+        },
     }
 
-    def _run_task(self):
-        repo = self.get_repo()
+    def _init_options(self, kwargs):
+        super(MergeBranch, self)._init_options(kwargs)
 
-        commit = self.options.get('commit', self.project_config.repo_commit)
-        branch_prefix = self.options.get('branch_prefix', self.project_config.project__git__prefix_feature)
-        source_branch = self.options.get('source_branch', self.project_config.project__git__default_branch)
+        if 'commit' not in self.options:
+            self.options['commit'] = self.project_config.repo_commit
+        if 'branch_prefix' not in self.options:
+            self.options['branch_prefix'] = self.project_config.project__git__prefix_feature
+        if 'source_branch' not in self.options:
+            self.options['source_branch'] = self.project_config.project__git__default_branch
+        if 'children_only' not in self.options:
+            self.options['children_only'] = False
+        elif self.options['children_only'] in [True, 'True']:
+            self.options['children_only'] = True
+
+    def _run_task(self):
+        self.repo = self.get_repo()
 
-        head_branch = repo.branch(source_branch)
+        head_branch = self.repo.branch(self.options['source_branch'])
         if not head_branch:
-            message = 'Branch {} not found'.format(source_branch)
+            message = 'Branch {} not found'.format(self.options['source_branch'])
             self.logger.error(message)
             raise GithubApiNotFoundError(message)
 
         # Get existing pull requests targeting a target branch
-        existing_prs = []
-        for pr in repo.iter_pulls(state='open'):
-            if pr.base.ref.startswith(branch_prefix):
-                existing_prs.append(pr.base.ref)
-        
-        targets = []
-        for branch in repo.iter_branches():
-            if not branch.name.startswith(branch_prefix):
-                self.logger.info('Skipping branch {}: does not match prefix {}'.format(branch.name, branch_prefix))
+        self.existing_prs = []
+        for pr in self.repo.iter_pulls(state='open'):
+            if pr.base.ref.startswith(self.options['branch_prefix']):
+                self.existing_prs.append(pr.base.ref)
+       
+        # Create list and dict of all target branches 
```

Comment:
We're looping over a list of branches twice in a row (L55, L73). Could we instead do it all in one loop, adding parent/child info along the way?

Reply:
You have to identify all children first and then go searching for their parents.  You might have a child early in the list whose parent you haven't get iterated to in the loop.  So you need a loop to build a list of possible children and possible parents and then a list to loop through the child to try to find a parent.